app/helpers/uploads_helper.rb

```diff
@@ -0,0 +1,35 @@
+require 'fileutils'
+require 'audit'
+require 'open3'
+
+module UploadsHelper
+  class Postprocessor
+    def self.process(f)
+      if self.respond_to?(File.extname(f)[1..-1])
+        self.method(File.extname(f)[1..-1]).call(f)
+      end
+    end
+
+    def self.rkt(f)
+      output, err, status = Open3.capture3("xvfb-run", "-a", "--server-num", "1",
```

Comment:
fast enough: a couple seconds at most, on a short-ish file with ~10 pictures in it.  (In terms of raw numbers of bytes, a few images >> many lines of text, so 10 pictures is a good proxy for a larger file.)  It's absolutely essential for Racket submissions, though, since the `#wxme` format of fancy files (containing embedded images, big comment boxes, etc) is illegible and doesn't lend itself to the inline commenting feature that makes BN comments so user-friendly.  I thought about sticking this in a delayed_job, but I think it's sufficiently critical-path that I'm ok delaying the appearance of the submission until this is processed.